lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -2156,9 +2156,12 @@ public function walkBetweenExpression($betweenExpr)
     public function walkLikeExpression($likeExpr)
     {
         $stringExpr = $likeExpr->stringExpression;
-        $leftExpr   = is_string($stringExpr) && isset($this->queryComponents[$stringExpr]['resultVariable'])
-            ? $this->walkResultVariable($stringExpr)
-            : $stringExpr->dispatch($this);
+        if (is_string($stringExpr)) {
+            assert(isset($this->queryComponents[$stringExpr]['resultVariable']));
```

Comment:
I've moved the `isset()` check from the condition into an `assert()`. In the current code, if that check would return `false`, we would call `dispatch()` on a string which would cause an error. This is why I believe that this check never fails. Alternatively, I could just remove the `isset()` check completely.

Reply:
Does it mean we can now have an error case where `$this->queryComponents[$stringExpr]` does not exist? Will this run into an error too?

Reply:
The previous code assumed that this case would never happen. But we could be extra cautious here and throw an exception. After all, it's a public method, so validating arguments is a sane thing to do.

Reply:
I'd prefer an exception too. Having a second opinion would be advisable.

Reply:
I'm throwing an exception instead now.